src/Bicep.Core/Semantics/DeclarationVisitor.cs

```diff
@@ -137,6 +137,14 @@ public override void VisitOutputDeclarationSyntax(OutputDeclarationSyntax syntax
             DeclareSymbol(symbol);
         }
 
+        public override void VisitAssertDeclarationSyntax(AssertDeclarationSyntax syntax)
+        {
+            base.VisitAssertDeclarationSyntax(syntax);
+
+            var symbol = new AssertSymbol(this.context, syntax.Name.IdentifierName, syntax);
+            DeclareSymbol(symbol);
```

Comment:
Since this feature was added to ARM under a feature flag, it should be under a feature flag in Bicep, too. This would be a good place to check if the feature has been enabled.

Reply:
+1 on having it behind a feature flag. @jeskew types are also behind a feature flag. Where did you block them when the feature was disabled?

Reply:
Types are blocked in `DeclaredTypeManager`. Type syntax can be used in `param` and `output` statements, so blocking `type` statements alone wasn't sufficient.

Reply:
I'm curious why we didn't block at them in the parser.

Reply:
Type expressions weren't a discrete category of syntax in the first implementation, so the parser didn't always have enough context to know if a given expression was a type only allowed with the flag set or if it was a value that should be allowed in all templates. (This changed because I wanted to allow decorators on properties of object types, but by then all the flag checking plumbing was in place.)

Reply:
I was thinking only of the `type` top-level declaration, but you'd also have to block usage of type expressions. Makes sense.